common/elasticsearch/esql/README.md

```diff
@@ -0,0 +1,135 @@
+}
+// with the 2 policies , converted dsl is equivalent to
+// "SELECT myColA FROM myTable WHERE myColB < 10 AND dateTime = '1561678568048000000'
+// in which the time is in unix nano format
+e := NewESql()
+e.SetReplace(myFilter1, myReplace)     // set up filtering policy
+e.SetProcess(myFilter2, myProcess)     // set up process policy
+dsl, _, err := e.ConvertPretty(sql)    // convert sql to dsl
+if err == nil {
+    fmt.Println(dsl)
+}
+~~~~
+For Cadence usage, refer to [this](https://github.com/jysui123/esql/blob/master/cadenceDevReadme.md) link.
+
+
+## Testing
+We are using elasticsearch's SQL translate API as a reference in testing. Testing contains 3 basic steps:
+- using elasticsearch's SQL translate API to translate sql to dsl
+- using our library to convert sql to dsl
+- query local elasticsearch server with both dsls, check the results are identical
+
+However, since ES's SQL api is still experimental, there are many features not supported well. For such queries, testing is mannual.
+
+Features not covered yet:
+- `LIKE`, `REGEXP` keyword: ES V6.5's sql api does not support regex search but only wildcard (only support shell wildcard `%` and `_`)
+- some aggregations are tested by manual check since ES's sql api does not support them well
+
+To run test locally:
+- download elasticsearch v6.5 (optional: kibana v6.5) and unzip them
+- run `chmod u+x start_service.sh test.sh`
+- run `./start_service.sh <elasticsearch_path> <kibana_path>` to start a local elasticsearch server (by default, elasticsearch listens port 9200, kibana listens port 5600)
+- run `python gen_test_date.py -dmi 1 1000 20` to insert 1000 documents to the local es
+- run `./test.sh TestSQL` to run all the test cases in `/testcases/sqls.txt`
+- generated dsls are stored in `dslsPretty.txt` for reference
+
+To customize test cases:
+- modify `testcases/sqls.txt`
+- run `python gen_test_date.py -h` for guides on how to insert custom data into your lcoal es
+- invalid query test cases are in `testcases/sqlsInvalid.txt`
+
+
+## ES V2.x vs ES V6.5
+|Item|ES V2.x|ES v6.5|
+|:-:|:-:|:-:|
+|missing check|{"missing": {"field": "xxx"}}|{"must_not": {"exist": {"field": "xxx"}}}|
+|group by multiple columns|nested "aggs" field|"composite" flattened grouping|
+
+
+## Attentions
+- We assume all the text are store in type `keyword`, i.e., ES does not break the text into separated words. And we use tag `term` to match the whole text rather than a word in the text.
+- `must_not` in ES does not share the same logic as `NOT` in sql
+- If you want to apply aggregation on some fields, they should be in type `keyword` in ES (set type of a field by put mapping to your table)
+- `COUNT(colName)` will include documents w/ null values in that column in ES SQL API, while in esql we exclude null valued documents
+- ES SQL API and esql do not support `SELECT DISTINCT`, but we can achieve the same result by `SELECT * FROM table GROUP BY colName`
```

Comment:
I think instead of "but we can...", it would be better to say "a work around is to use", otherwise it makes me think you added dedup logic in group by.